### ISSUE TITLE
/usr/bin/env: ‘node\r’: No such file or directory

### ISSUE THREAD

**Issue by pawangspandey** [Azure/iothub-explorer#6]
I can not able to run the iothub-explorer cli after successful installation.
I am using 
Node v6.9.1.
NPM 3.10.8.
OS Ubuntu xenial 16.04.1 LTS

**Comment by pierreca** [comment]
@pawangspandey 
- how did you install iothub-explorer?
- how are you running it?

Also can you run the following command in a terminal and verify that it actually starts a node.js process?
```
/usr/bin/env node
```

**Comment by pawangspandey** [comment]
I am able to start a node.js process with the following command.
`/usr/bin/env node`

I used the following command to install iothub-explorer
`sudo npm install -g iothub-explorer`

I am using `iothub-explorer help` command to get addition command list but getting error `/usr/bin/env: ‘node\r’: No such file or directory`

**Comment by pierreca** [comment]
So the help command works fine, but not the others? or do you get this error while running the help command?

can you please paste the whole console log?

**Comment by pawangspandey** [comment]
No, I can not able to run any of the commands which are listed on homepage of this repo. even I cannot able to run `iothub-explorer -V`.
Every Time I got this error `/usr/bin/env: ‘node\r’: No such file or directory` printed on terminal.

**Comment by pawangspandey** [comment]
Here is the screenshot of my terminal.


![iothub error](https://cloud.githubusercontent.com/assets/12795548/22238115/a069c058-e235-11e6-80a6-1e323c220216.png)

**Comment by pierreca** [comment]
looks like the line endings are off...

Windows and Linux use different line endings. (`\r\n` vs just `\n`). The weird part is, this new version runs fine on Ubuntu for me... so I wonder if it's a packaging issue, or a system configuration thing. 

can you run the following:
```
which iothub-explorer
```
That should tell you where a symlink to iothub-explorer is installed. go there then:
```
ls -l iothub-explorer
```
This should give you the real path to where iothub-explorer is installed (the js files). Go there, then:
```
file iothub-explorer.js
```
That should tell you what line endings the file has. For example I have:
```
iothub-explorer.js: Node.js script, ASCII text executable
```

Just for kicks I ran `unix2dos` to change the line endings, and at that point I got an error in my terminal (not the one you get, but I'm betting it's the same kind of problem). After running `unix2dos` I get this:
```
iothub-explorer.js: Node.js script, ASCII text executable, with CRLF line terminators
```

So if you get that last part (`with CRLF line terminators`) I would encourage you to run the opposite, `dos2unix` and see if that fixes it (at least for `iothub-explorer -V`). 

I'm going to look at the package. Running those steps on your machine would help narrow down the issue.

**Comment by pierreca** [comment]
Alright, I changed our publication mechanism and published version 1.1.5 of iothub-explorer on npm. Can you please try installing this and let me know if that fixed it for you?

**Comment by pawangspandey** [comment]
thanks, @pierreca , the current published version 1.1.5 solved the issue.
